### PR TITLE
Controller/rest integration pdf to image

### PR DESCRIPTION
**Pull request by MadayAlcala**
Hi guys, this is the implementation for pdf to image integration, please review it.

### CODE REVIEW

**Review by JosueRodriguezGarcia — COMMENTED** [review]
Ok Maday, your code is fine.

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 1]
```diff
@@ -45,40 +48,43 @@
...
-            @RequestParam(value = "width", defaultValue = "") String width, 
-            @RequestParam(value = "height", defaultValue = "") String height) {
-        String fileName = uploadService.storeFile(file);
+    @PostMapping("/uploadPdf")
+    public pdfResponse upload(@RequestParam("pdf") MultipartFile pdf,
+                              @RequestParam(value = "name", defaultValue = "") String name,
+                              @RequestParam(value = "dpi", defaultValue = "") String dpi,
+                              @RequestParam(value = "extension", defaultValue = "") String ext,
+                              @RequestParam(value = "formatColor", defaultValue = "") String formatColor){
+        String pdfName = uploadService.storeFile(pdf);
```
Comment (JosueRodriguezGarcia) [inline src/main/java/com/fundation/webservice/controller/Controller.java]: Maday is necesary this space?

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 2]
```diff
@@ -45,40 +48,43 @@
...
         String fileDownloadUri = ServletUriComponentsBuilder.fromCurrentContextPath()
-            .path("/download/")
-            .path(fileName)
-            .toUriString();
+                .path("/download/")
+                .path(name + ".zip")
+                .toUriString();
+
+        //creating a new folder for the converted images
+        new File("C:/_pg/tmp/conversions/" + name + "/").mkdirs();
```
Comment (JosueRodriguezGarcia) [inline src/main/java/com/fundation/webservice/controller/Controller.java]: Maday is necesary this space?

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 3]
```diff
@@ -45,40 +48,43 @@
...
+        criterion.setName(name);
+        criterion.setDpi(new Integer(dpi));
+        criterion.setExt(ext);
+        criterion.setFormatColor(formatColor);
+        ConvertPdfToImage cuento = new ConvertPdfToImage(criterion);
+        cuento.convert();
+
+        //This line compresses the folder with images in a zip file
+        FolderZipped.zipFolder(name);
+
```
Comment (JosueRodriguezGarcia) [inline src/main/java/com/fundation/webservice/controller/Controller.java]: Maday is necesary this space?

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 4]
```diff
@@ -45,40 +48,43 @@
...
+        return new pdfResponse(pdfName, fileDownloadUri, pdf.getContentType(),
+                pdf.getSize(), name, dpi, ext, formatColor);
     }
 
+
```
Comment (JosueRodriguezGarcia) [inline src/main/java/com/fundation/webservice/controller/Controller.java]: Maday is necesary this space?

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 5]
```diff
@@ -45,40 +48,43 @@
...
-            height);
+        //Converting pdf into images
+        CriteriaPdfToImage criterion = new CriteriaPdfToImage();
+        criterion.setSrcPath("C:\\_pg\\tmp\\uploads\\" + pdfName);
+        criterion.setDestPath("C:\\_pg\\tmp\\conversions\\"+name + "\\");
+        criterion.setName(name);
+        criterion.setDpi(new Integer(dpi));
+        criterion.setExt(ext);
+        criterion.setFormatColor(formatColor);
+        ConvertPdfToImage cuento = new ConvertPdfToImage(criterion);
```
Comment (JosueRodriguezGarcia) [inline src/main/java/com/fundation/webservice/controller/Controller.java]: Cuento?

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 1 above)
Comment (MadayAlcala) [reply src/main/java/com/fundation/webservice/controller/Controller.java]: did it

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 2 above)
Comment (MadayAlcala) [reply src/main/java/com/fundation/webservice/controller/Controller.java]: solved

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 3 above)
Comment (MadayAlcala) [reply src/main/java/com/fundation/webservice/controller/Controller.java]: solved

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 4 above)
Comment (MadayAlcala) [reply src/main/java/com/fundation/webservice/controller/Controller.java]: solved

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 5 above)
Comment (MadayAlcala) [reply src/main/java/com/fundation/webservice/controller/Controller.java]: fixed

**Comment by JosueRodriguezGarcia** [comment]
Ok, great Job.

**Pull request merged**